### PR TITLE
feat: support aggr filter chnages in zapiperf

### PR DESCRIPTION
**Pull request by Hardikl**
New `volume_aggr_labels` metric,

for Flexgroup,
image

for Flexvol,
image

other existing metrics,
for Flexgroup,
image

for Flexvol,
image

### CODE REVIEW

File: cmd/collectors/zapiperf/plugins/volume/volume.go [hunk 1]
```diff
@@ -32,51 +33,65 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
-	metric, err := data.NewMetricFloat64(metricName, metricName)
+	flexgroupAggrsMap := make(map[string]*set.Set)
+	// new metric would be volume_aggr_labels
+	metricName := "aggr_labels"
+	volumeAggrmetric := matrix.New(".Volume", "volume", "volume")
+	volumeAggrmetric.SetGlobalLabels(data.GetGlobalLabels())
+
+	metric, err := volumeAggrmetric.NewMetricFloat64(metricName)
 	if err != nil {
 		me.Logger.Error().Stack().Err(err).Msg("add metric")
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/plugins/volume/volume.go:44]: `Stack()` not needed for any of the log msgs in this class. The msg and file:line is sufficient for root cause and troubleshooting

File: cmd/collectors/zapiperf/plugins/volume/volume.go [hunk 2]
```diff
@@ -32,51 +33,65 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
+
+			if volumeAggrmetric.GetInstance(key) == nil {
+				flexgroupInstance, _ := volumeAggrmetric.NewInstance(key)
+				flexgroupInstance.SetLabels(i.GetLabels().Copy())
+				flexgroupInstance.SetLabel("volume", match[1])
+				// Flexgroup don't show any node
+				flexgroupInstance.SetLabel("node", "")
+				flexgroupInstance.SetLabel("style", "flexgroup")
+				flexgroupAggrsMap[key] = set.New()
+				flexgroupAggrsMap[key].Add(i.GetLabel("aggr"))
```
Comment (cgrinds) [inline cmd/collectors/zapiperf/plugins/volume/volume.go:75]: minor: this line can be moved to line 83 and then the else clause at 79 removed

File: cmd/collectors/zapiperf/plugins/volume/volume.go (hunk 1 above)
Comment (Hardikl) [reply cmd/collectors/zapiperf/plugins/volume/volume.go]: done

**Comment by Hardikl** [comment]
Tested against 9.12 cluster - 10.193.48.154,

new metrics:
image

existing metrics:
image

File: cmd/collectors/restperf/plugins/volume/volume.go [hunk 3]
```diff
@@ -41,10 +56,33 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 				fg.SetLabel("node", "")
 				fg.SetLabel("style", "flexgroup")
 			}
+
+			if volumeAggrmetric.GetInstance(key) == nil {
+				flexgroupInstance, _ := volumeAggrmetric.NewInstance(key)
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go]: done

File: cmd/collectors/restperf/plugins/volume/volume.go [hunk 4]
```diff
@@ -41,10 +56,33 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
...
+				}
+			}
+			flexgroupAggrsMap[key].Add(i.GetLabel("aggr"))
 			i.SetLabel("style", "flexgroup_constituent")
 			i.SetExportable(false)
 		} else {
 			i.SetLabel("style", "flexvol")
+			key := i.GetLabel("svm") + "." + i.GetLabel("volume")
+			if volumeAggrmetric.GetInstance(key) == nil {
+				flexvolInstance, _ := volumeAggrmetric.NewInstance(key)
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go]: done

File: cmd/collectors/restperf/plugins/volume/volume.go [hunk 5]
```diff
@@ -24,6 +26,19 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 
 	re := regexp.MustCompile(`^(.*)__(\d{4})$`)
 
+	flexgroupAggrsMap := make(map[string]*set.Set)
+	// new metric would be volume_aggr_labels
+	metricName := "aggr_labels"
+	volumeAggrmetric := matrix.New(".Volume", "volume", "volume")
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go]: I have checked the metrics from template are populated as-is., But still for safer side, changed identifier from `volume` to `volume_aggr` and new metric name would be same as before.

File: cmd/collectors/restperf/plugins/volume/volume.go [hunk 6]
```diff
@@ -41,10 +56,35 @@ func (me *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 				fg.SetLabel("node", "")
 				fg.SetLabel("style", "flexgroup")
 			}
+
+			flexgroupInstance, err := volumeAggrmetric.NewInstance(key)
+			if err == nil {
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go:61]: Nope, when first constituent came, we should create new instance for flexgroup, but for next constituents, we don't need to create, we just update aggr info in that instance.

Error would be possible only for flexvol as it should not create new instance for same key, and it's logged at line 79.

[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go:61]: NewInstance function only error out when duplicate key found

[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/volume/volume.go:61]: Okie, 

Earlier have implemented the same :) 
image

Will change for flexgroup this way, as flexvol would err out if duplicate instance

**Pull request merged**